v1/scanner.go

```diff
@@ -52,7 +53,16 @@ func transformSyntacticError(err error) error {
 		if serr.Err == io.ErrUnexpectedEOF {
 			serr.Err = errUnexpectedEnd
 		}
-		return &SyntaxError{Offset: serr.ByteOffset, msg: serr.Err.Error()}
+		msg := serr.Err.Error()
+		if i := strings.Index(msg, " (expecting"); i >= 0 && !strings.Contains(msg, " in literal") {
+			msg = msg[:i]
+		}
+		msg = strings.ReplaceAll(msg, "object name", "object key")
+		msg = strings.ReplaceAll(msg, "at start of value", "looking for beginning of value")
+		msg = strings.ReplaceAll(msg, "at start of string", "looking for beginning of object key string")
+		msg = strings.ReplaceAll(msg, "after object value", "after object key:value pair")
+		msg = strings.ReplaceAll(msg, "in number", "in numeric literal")
```

Comment:
did you consider a strings.Replacer?

Reply:
👀 . This will be the first time I have ever used `strings.Replacer`.